src/foam/u2/ActionView.js

```diff
@@ -155,13 +155,17 @@ foam.CLASS({
     function click(e) {
       try {
         if ( this.action && this.action.confirmationView && this.buttonState == this.ButtonState.NO_CONFIRM ) {
-          (async () => {
-            this.ctrl.add(this.ConfirmationModal.create({
-              primaryAction: this.action,
-              data: this.data,
-              title: this.action.confirmationView().title || this.action.label + ' ' + await this.data.toSummary() + '?'
-            }).add(this.action.confirmationView().body || this.CONFIRM_MSG + ' ' + this.action.label.toLowerCase() + ' ' + await this.data.toSummary() + '?'));
-          })();
+          let modal = this.action.confirmationView(this.__subContext__, this.data)
+          if (!modal) { this.action && this.action.maybeCall(this.__subContext__, this.data) }
```

Comment:
Spacing should be:
if ( ! modal ) { ...
See the foam style guide: https://github.com/kgrgreer/foam3/blob/development/doc/guides/StyleGuide.md